src/types/utils.ts

```diff
@@ -0,0 +1,5 @@
+export type OptionalExclude<T, Exclusion extends keyof T> = Required<Pick<T, Exclusion>> & Partial<Omit<T, Exclusion>>;
```

Comment:
<img width="388" alt="Screen Shot 2021-12-17 at 12 27 42 PM" src="https://user-images.githubusercontent.com/86971252/146584179-86ef9249-9af7-4c4a-b3c5-a2ef0decac71.png">

Reply:
This utility type makes every except for the excluded property in the given type optional

Reply:
👍 just j/k